.github/workflows/tests.yml

```diff
@@ -8,17 +8,26 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
 
-      - name: Setup node
+      - name: Cache dependencies
+        uses: actions/cache@v3
+        id: npm-cache
+        with:
+          key: npm-cache-${{ hashFiles('package-lock.json', 'package.json') }}
```

Comment:
Is this not already handled by `setup-node` using `cache: npm`? 🤔 

https://github.com/actions/setup-node#caching-global-packages-data

Reply:
Unlike `cache: npm` (only **~/.npm**) this one makes sure **./node_modules** is also cached

It means we can avoid `npm ci` entirely as we have an exact match in the cache. We'll only do this when we're 100% certain (via the `hashFiles()` match) but it's a huge performance win.

Should the hash not match due to package version changes, it's still good to cache **~/.npm** so that the 99% _unchanged_ packages can be reinstalled without hitting the network via https://registry.npmjs.com

Reply:
I've updated this line to remove `'package.json'` as we install using `npm ci` in this workflow